### PR TITLE
Phase2-hgx342B Investigating the difference between DDD and DD4hep geometries for HGCal partial wafers

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Investigating the difference between DDD and DD4hep geometries for HGCal partial wafers

#### PR validation:

Comparison of the results reveal a few diagnostics which need further clarification

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/Calo/src/HGCalNumberingScheme.cc [hunk 1]
```diff
@@ -91,18 +91,19 @@ uint32_t HGCalNumberingScheme::getUnitID(int layer, int module, int cell, int iz
...
     int cellU(0), cellV(0), waferType(-1), waferU(0), waferV(0);
+    bool debug(false);
     if (cell >= 0) {
       waferType = HGCalTypes::getUnpackedType(module);
       waferU = HGCalTypes::getUnpackedU(module);
       waferV = HGCalTypes::getUnpackedV(module);
       cellU = HGCalTypes::getUnpackedCellU(cell);
       cellV = HGCalTypes::getUnpackedCellV(cell);
     } else if (mode_ != HGCalGeometryMode::Hexagon8) {
+      debug = true;
```
Comment (civanch) [inline SimG4CMS/Calo/src/HGCalNumberingScheme.cc:102]: @bsunanda , is it what you need for master (debug = true) ?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by srimanob** [comment]
+Upgrade

Code for debugging, no change is expected in any workflows.

**Comment by rappoccio** [comment]
+1

**Pull request merged**